### PR TITLE
Configuration/setting rubocop

### PR DESCRIPTION
**Pull request by LeandroSoria2**
### What changes does this PR include?
**Description**
Add rubocop gem and settings 
**Summary**
- Add rubocop gem
- Run rubocop
- Add file code_analysis

**Trello**

[Code analysis tools](https://trello.com/c/Ln84wLni/24-code-analysis-tools)

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+inherit_mode:
+  merge:
+    - Exclude
+
+AllCops:
+  Exclude:
+    - bin/bundle
+
+Layout/LineLength:
+  Exclude:
```
Comment (JulianPasquale) [inline .rubocop.yml:14]: Maybe we can exclude all the migrations and configs here?

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    - db/migrate/20220518201152_devise_token_auth_create_users.rb
+    - db/migrate/20220601190711_change_user_email_not_null.rb
+    - db/migrate/20220603145240_change_user_gender_enum.rb
+    - rubocGemfile
```
Comment (JulianPasquale) [inline .rubocop.yml]: What is this file?

File: spec/requests/users/create_spec.rb [hunk 3]
```diff
@@ -79,7 +79,7 @@
     context 'when passwords don\'t match' do
       let(:password)              { 'shouldmatch' }
       let(:password_confirmation) { 'dontmatch' }
-      let(:new_user)              { User.find_by(email: email) }
+      let(:new_user)              { User.find_by(email:) }
```
Comment (JulianPasquale) [inline spec/requests/users/create_spec.rb]: hmm looks like this change was not needed right? Probably this test is broken rn

Comment (JulianPasquale) [reply spec/requests/users/create_spec.rb]: Noticed this pattern in a lot of test. You need to add the variable back (like `email: email`)

**Review by sebastiancaraballo — COMMENTED** [review]
Please add Github Actions, here is an [example](https://github.com/rootstrap/rails_api_base/blob/master/.github/workflows/ci.yml).

File: .rubocop.yml (hunk 2 above)
Comment (LeandroSoria2) [reply .rubocop.yml]: I made mistype, I fixed in last commit

**Review by sebastiancaraballo — APPROVED** [review]

File: .rubocop.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+AllCops:
+  Exclude:
+    - bin/bundle
+
+Layout/LineLength:
+  Exclude:
+    - db/migrate/20220518201152_devise_token_auth_create_users.rb
+    - db/migrate/20220601190711_change_user_email_not_null.rb
+    - db/migrate/20220603145240_change_user_gender_enum.rb
```
Comment (JulianPasquale) [inline .rubocop.yml]:
```suggestion
    - db/migrate/*
```

**Review by JulianPasquale — APPROVED** [review]

File: spec/requests/users/create_spec.rb (hunk 3 above)
Comment (LeandroSoria2) [reply spec/requests/users/create_spec.rb]: fixed in last commit

File: .rubocop.yml (hunk 4 above)
Comment (LeandroSoria2) [reply .rubocop.yml]: fixed in last commit

**Pull request merged**